### PR TITLE
Task4 Show all articles

### PR DESCRIPTION
**Pull request by stevenwu99**
Good morning!

I have complete Task 4 for showing all articles.
Please check.
thank you

### CODE REVIEW

File: src/components/ArticleCard.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const articleCard = ({articles}) => {
+    return (
+        <div>
+            {
+                 articles.map((article) =>{
+                 return  ( 
+                       <div key = {article.article_id}>
+                        <label name = "articleId">Id: {article.article_id}</label> <br/>
```
Comment (roisinoreilly) [inline src/components/ArticleCard.jsx]: I don't think the article ID is relevant to the user so I would remove it to keep your article card neater

File: src/components/ArticleCard.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                        <label name = "articletitle">Title: {article.title}</label> <br/>
+                        <label name = "articleauthor">Author: {article.author}</label> <br/>
+                        <label name = "article.topic">Topic: {article.topic}</label> <br/>
+                        <label name = "articlecreatedat">Created_at: {article.created_at}</label> <br/>
+                        <img src = {article.article_img_url}  alt={'this picture for' + article.title} /> <br/>
```
Comment (roisinoreilly) [inline src/components/ArticleCard.jsx]: alt text shouldn't include that something is an image or a picture, I would just use the article title here

File: src/components/ArticleCard.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const articleCard = ({articles}) => {
+    return (
+        <div>
+            {
+                 articles.map((article) =>{
+                 return  ( 
+                       <div key = {article.article_id}>
+                        <label name = "articleId">Id: {article.article_id}</label> <br/>
+                        <label name = "articletitle">Title: {article.title}</label> <br/>
```
Comment (roisinoreilly) [inline src/components/ArticleCard.jsx]: label isn't the most appropriate HTML element here, it's usually used to with forms such as `<input>`, `<textarea>`, or `<select>`, can you change these to header or p tags

File: src/components/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const Home = () => {
+    return (
+        <div>
```
Comment (roisinoreilly) [inline src/components/Home.jsx]: I don't think the `<div>` or `<section>` provide any semantic meaning to the component

File: src/components/ShowArticles.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { useState,useEffect} from "react";
+import {getData,showLoading} from "../utilities/API";
+import ArticleCard from "./ArticleCard";
+
+const  ShowArticles = ({categoryFilter}) => {
+    
+    const [articles,setArticles] = useState([]);
+    const [isLoading,setIsLoading] = useState(true)
+   
+    let  searchUrl = "/articles";
```
Comment (roisinoreilly) [inline src/components/ShowArticles.jsx:8]: does this need to be let?

File: src/utilities/API.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+export const getData = (searchUrl) => {
+       return db.get(searchUrl)
+        .then(({data}) => {
+            return data;
+         })
+    }
+
+export const showLoading = (isLoading) => {
+    if (isLoading) {
```
Comment (roisinoreilly) [inline src/utilities/API.jsx]: hmmm this isn't displaying for me on the articles page

File: src/components/NavBar.jsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+const NavBar = () => {
+  return (
+    <div>
```
Comment (roisinoreilly) [inline src/components/NavBar.jsx]: this should be surrounded with `<nav>` to allow people using screen readers to navigate your site

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
great start Steven! I can see you've made considerations for a nice mobile first design and UI/UX and your components are appropriate and a good size. I just have a few changes I'd like you to make

- you've made a loading component which is great, however I can't actually see it when I throttle the network, can you double check this works
- your timestamp isn't very user friendly, can you think about how you would manipulate the timestamp data to something more human readable or look into some packages you can use that will do this for you
- I don't think the user needs to see the article ID in the article cards, I would remove that
- be careful with your semantic HTML and which elements you choose. Can you remove all uses of `<label>` as this should be used for surrounding a form element not just text. Can you also implement a heading structure in your articles and `<nav>` tags in your navbar component for accessibility

once you've implemented these changes can you send us another PR :)

File: src/components/ArticleCard.jsx (hunk 1 above)
Comment (stevenwu99) [reply src/components/ArticleCard.jsx]: done

File: src/components/ArticleCard.jsx (hunk 2 above)
Comment (stevenwu99) [reply src/components/ArticleCard.jsx]: done

File: src/components/ArticleCard.jsx (hunk 3 above)
Comment (stevenwu99) [reply src/components/ArticleCard.jsx]: done

File: src/components/Home.jsx (hunk 4 above)
Comment (stevenwu99) [reply src/components/Home.jsx]: done

File: src/components/ShowArticles.jsx (hunk 5 above)
Comment (stevenwu99) [reply src/components/ShowArticles.jsx:8]: done

File: src/utilities/API.jsx (hunk 6 above)
Comment (stevenwu99) [reply src/utilities/API.jsx]: Done

File: src/components/NavBar.jsx (hunk 7 above)
Comment (stevenwu99) [reply src/components/NavBar.jsx]: Done

**Review by yalicee — APPROVED** [review]
Thank you for making the changes :)

**Pull request merged**